### DISCUSSION TITLE
How do I get the id of a custom node inside the custom node component?

### DISCUSSION THREAD

**Discussion by vothvovo** [xyflow/xyflow#3037] [Q&A] [answered]
I want to implement a selection highlighting system.
Therefore, I think, I need to go into the custom node component and do an onClick with the nodes id.

My current solution looks something like this:
<img width="685" alt="image" src="https://user-images.githubusercontent.com/20894472/236602355-9ea59c71-d289-4109-8004-fbc456f766a0.png">
using zustand and tailwind here. I put the id of the node as a duplicate in its data, so that it is available to me in the custom node component.

**Comment by bcakmakoglu** [comment, marked as answer]
The id is passed as a node prop

```tsx
const CustomNode = ({ id, data }: NodeProps) => {
   // ...
}
```
